launcher/ui/pages/global/LauncherPage.cpp

```diff
@@ -410,6 +413,10 @@ void LauncherPage::loadSettings()
         }
     }
 
+    // Toolbar/menu bar settings (not applicable if native menu bar is present)
+    ui->toolsBox->setVisible(!QMenuBar().isNativeMenuBar());
```

Comment:
hiding this might be fine on macOS, as you can't get rid of the native menubar, but on DEs like KDE Plasma the user could remove the global menu on the fly, which would suddenly show this option to them.

I would prefer disabling it instead of hiding it. Maybe on macOS you can hide it as well.
So:
```suggestion
    ui->toolsBox->setEnabled(!QMenuBar().isNativeMenuBar());
#ifdef Q_OS_MACOS
    ui->toolsBox->setVisible(!QMenuBar().isNativeMenuBar());
#endif
```

Reply:
By the way, Qt docs suggest that the KDE Plasma global menu wouldn't be supported as "native", and thus the option would always show. I still applied this change though because I'm not sure if something similar is possible on Unity(/other DEs?), and I don't have easy means to test it right now.